modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -5247,11 +5264,15 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
             codegenParameter.style = parameter.getStyle().toString();
             codegenParameter.isDeepObject = Parameter.StyleEnum.DEEPOBJECT == parameter.getStyle();
             codegenParameter.isMatrix = Parameter.StyleEnum.MATRIX == parameter.getStyle();
+        } else if (openAPI != null && getMajorSpecVersion(getOriginalSpecVersion(openAPI)) >= 3) {
+            // Properly sets the default style for oas V3
+            codegenParameter.style = defaultV3Style;
```

Comment:
Like in lines 5264-5266 above, when setting `codegenParameter.style`, we should also set `codegenParameter.isDeepObject` and `codegenParameter.isMatrix` accordingly. To fix this and to also avoid such problems in the future, could you please make `CodegenParameter.style` private, add a Getter and Setter, and in the Setter also set `isDeepObject` and `isMatrix` accordingly?